src/coresyf_image_crop/coresyf_image_crop.py

```diff
@@ -52,38 +48,30 @@ def get_shapefile_polygon_extent(data_source):
     return ogr_polygon
 
 
-def get_shapefile_crs(data_source):
+def get_datasource_epsg(data_source):
     '''
-    Retrieves the ESPG Code of the shapefile CRS (Coordinate Reference System).
+    Retrieves the ESPG Code of a GDAL/OGR datasource CRS (Coordinate Reference
+    System).
 
-    data_source: must be the result of GDAL Open applied to the respective OGR
-    driver.
+    data_source: must be the result of GDAL/OGR Open applied to the respective
+                 data.
     '''
-    try:
+    if isinstance(data_source, ogr.DataSource):
+        name = data_source.GetName()
         in_layer = data_source.GetLayer()
         spatial_ref = in_layer.GetSpatialRef()
-        ident = Sridentify(prj=spatial_ref.ExportToWkt())
-        epsg_code = ident.get_epsg()
-    except Exception as epsg_exception:
-        print(epsg_exception)
-        sys.exit("Error. Unable to get ESPG code of grid shapefile.")
-    return epsg_code
-
-
-def get_raster_crs(data_source):
-    '''
-    Retrieves the ESPG Code of the raster CRS (Coordinate Reference System).
+        projection_wkt = spatial_ref.ExportToWkt()
+    elif isinstance(data_source, gdal.Dataset):
+        name = data_source.GetDescription()
+        projection_wkt = data_source.GetProjectionRef()
+    else:
+        raise TypeError("Input is not of ogr.DataSource or gdal.Dataset type!")
 
-    data_source: must be the result of GDAL Open applied to the respective GDAL
-    driver.
-    '''
-    try:
-        projection = data_source.GetProjectionRef()
-        ident = Sridentify(prj=projection)
-        epsg_code = ident.get_epsg()
-    except Exception as epsg_exception:
-        print(epsg_exception)
-        sys.exit("Error. Unable to get ESPG code of raster.")
+    sridentify_obj = Sridentify(prj=projection_wkt)
+    epsg_code = sridentify_obj.get_epsg()
+    if not epsg_code:
```

Comment:
We should be very careful with these kind of conditions. Remember that if `epsge_code` is 0 this condition also evaluates to `false`. In general, is a better approach to do explicit comparisons like `epsg_code is not None`.

I think this is not a problem here, I suppose 0 is not a valid _ESPG code_. Please confirm this.